src/nnet3/nnet-diagnostics.h

```diff
@@ -36,7 +36,12 @@ struct SimpleObjectiveInfo {
   double tot_objective;
   SimpleObjectiveInfo(): tot_weight(0.0),
                          tot_objective(0.0) { }
+};
 
+struct PerDimObjectiveInfo : SimpleObjectiveInfo {
```

Comment:
you should say "public" here, and no space before ':'.
And there should be some documentation of the members' meanings.
Is the constructor necessary? Surely the default would be sufficient.